sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/XmlExternalEntityShouldNotBeParsed.cs

```diff
@@ -139,7 +139,7 @@ private void VerifyXPathDocumentConstructor(SyntaxNodeAnalysisContext context)
 
         private static bool IsXPathDocumentSecureByDefault(NetFrameworkVersion version) =>
             // XPathDocument is secure by default starting with .Net 4.5.2
-            version == NetFrameworkVersion.After452 || version == NetFrameworkVersion.Unknown;
+            version >= NetFrameworkVersion.After452 || version == NetFrameworkVersion.Unknown;
```

Comment:
This will need to be rewritten a little, after addressing the comment in `NetFrameworkVersions`. 

The semantic issue in this usage is, that the enum's values are implicit and therefore should not be use for "greater then" comparation. Let's use a switch expression here, just like `ConstructorIsSafe` does.